File: app/src/main/java/com/noticias_now/ui/fragment/BaseFragment.java
package com.noticias_now.ui.fragment;


import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.noticias_now.app.ApsNoticiasApp;
import com.noticias_now.ui.activity.BaseActivity;
import com.noticias_now.model.UsuarioModel;

/**
 * A simple {@link Fragment} subclass.
 */
public class BaseFragment extends Fragment {

    protected UsuarioModel usuario;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        usuario = ApsNoticiasApp.getInstance().getUsuario();
    }

    public void showProgressDialog(boolean isShow, String mensagem) {
        try {
            if (getActivity() instanceof BaseActivity) {
                getBaseActivity().showProgressDialog(isShow, mensagem);
            }
        } catch (Exception ex) {

        }
    }

    public BaseActivity getBaseActivity() {
        return ((BaseActivity) getActivity());
    }

    public void openActivity(Class<?> classOpen, String param, Object value) {
        try {
            getBaseActivity().openActivity(getActivity(), classOpen, param, value);
        } catch (Exception ex) {
        }
    }

    public void openActivity(Class<?> openActivty){
        getBaseActivity().openActivity(getActivity(), openActivty);
    }

    public void openActivityForResult(Class<?> openActivity, int requestCode){
        getBaseActivity().openActivityForResult(openActivity, requestCode);
    }

    public void openActivityForResult(Class<?> openActivity, String param, Object object, int requestCode){
        getBaseActivity().openActivityForResult(openActivity, param, object, requestCode);
    }

}

File: common/src/main/java/br/com/ricarlo/common/ui/base/BaseFragment.kt
package br.com.ricarlo.common.ui.base

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.annotation.LayoutRes
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import androidx.fragment.app.Fragment

/**
 * A simple [Fragment] subclass.
 */
abstract class BaseFragment<T : ViewDataBinding> : Fragment() {

    @LayoutRes
    abstract fun getLayoutRes(): Int

    abstract fun initView(savedInstanceState: Bundle?)

    protected lateinit var binding: T

    override fun onCreateView(
            inflater: LayoutInflater,
            container: ViewGroup?,
            savedInstanceState: Bundle?
    ): View? {
        binding = DataBindingUtil.inflate(inflater, getLayoutRes(), container, false)
        return binding.apply {
            lifecycleOwner = viewLifecycleOwner
        }.root
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        initView(savedInstanceState = savedInstanceState)
    }

    protected fun handlerError(throwable: Throwable) {
        (activity as? BaseActivity<*>)?.handlerError(throwable)
    }

    open fun showLoading(message: String?) {
        (activity as? BaseActivity<*>)?.showLoading(message)
    }

    open fun hideLoading() {
        (activity as? BaseActivity<*>)?.hideLoading()
    }

    fun openActivity(classOpen: Class<*>, bundle: Bundle? = null) {
        (activity as? BaseActivity<*>)?.openActivity(classOpen, bundle)
    }
}